homeassistant/components/music_assistant/media_player.py

```diff
@@ -587,16 +587,25 @@ def _update_media_image_url(
         self, player: Player, queue: PlayerQueue | None
     ) -> None:
         """Update image URL for the active queue item."""
-        if queue is None or queue.current_item is None:
-            self._attr_media_image_url = None
-            return
-        if image_url := self.mass.get_media_item_image_url(queue.current_item):
+        # image_url is provided by an external source
+        image_url = (
+            player.current_media.image_url
+            if player.current_media and player.current_media.image_url
+            else None
+        )
```

Comment:
let's put this block second. So always prefer an active queue's current item image.

Reply:
I'm a bit confused… :) The active queue already is preferred. Are you requesting the block to be rewritten in a if/elif-fashion? If so, no problem for sure.

Reply:
I've rewritten the logic because the confusion already states that it wasn't clear enough. Hope this version fits your request.